frontend/packages/console-app/src/components/cloud-shell/CloudShellTerminal.tsx

```diff
@@ -94,6 +96,26 @@ const CloudShellTerminal: React.FC<CloudShellTerminalProps &
     };
   }, [username, workspaceName, workspaceNamespace, workspacePhase, t]);
 
+  // If the user is able to create a pod in the openshift-terminal namespace then they
+  // are a cluster admin
+  React.useEffect(() => {
+    checkAccess({
+      namespace: 'openshift-terminal',
+      verb: 'create',
+      resource: 'pods',
+    })
+      .then((resp) => {
+        if (resp.status.allowed) {
+          setAdmin(true);
+          return;
+        }
+        setAdmin(false);
+      })
+      .catch((e) => {
+        setInitError(e);
+      });
+  }, []);
```

Comment:
You need to capture if the rbac check is still pending show the `TerminalLoadingBox`.

If catching the error is an edge case, we could simplify to use `useAccessReview2`. Otherwise add a separate loading state.
```
  const [isAdmin, isAdminCheckLoading] = useAccessReview2({
      namespace: 'openshift-terminal',
      verb: 'create',
      resource: 'pods',
    });
```